clients/src/main/java/org/apache/kafka/common/telemetry/internals/MetricsCollector.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.telemetry.internals;
+
+/**
+ * A {@code MetricsCollector} is responsible for scraping a source of metrics and forwarding
+ * them to the given {@link MetricsEmitter}. For example, a given collector might be used to collect
+ * system metrics, Kafka metrics, JVM metrics, or other metrics that are to be captured, exposed,
+ * and/or forwarded.
+ *
+ * <p/>
+ *
+ * In general, a {@code MetricsCollector} implementation is closely managed by another entity
+ * (that entity is colloquially referred to as the "telemetry reporter") that will be in
+ * charge of its lifecycle via the {@link #start()} and {@link #stop()} methods. The telemetry
+ * reporter should ensure that the {@link #start()} method is invoked <i>once and only once</i>
+ * before calls to {@link #collect(MetricsEmitter)} are made. Implementations of {@code MetricsCollector}
+ * should allow for the corner-case that {@link #stop()} is called before {@link #start()},
+ * which might happen in the case of error on startup of the telemetry reporter.
```

Comment:
And what happens then? I suppose that the MetricsCollector remains STOPPED, but does `start()` fail?

Reply:
Yes metrics collector should remain stopped and start should fail because of terminating signal received via stop().